polaris-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -140,14 +161,23 @@
     transform var(--p-motion-duration-100) var(--p-motion-ease);
 
   #{$se23} & {
+    top: calc(-1 * var(--p-space-05));
+    left: calc(-1 * var(--p-space-05));
+    bottom: calc(-1 * var(--p-space-05));
+    right: calc(-1 * var(--p-space-05));
```

Comment:
The effect of these hard coded positions is to scale up the "indeterminate" svg from 20x20 -> 22x22. The svg's viewbox is only 20x20, so rendering it larger could make it blurry on some screens.

For example, here's how it rendered for me in Chrome on my MacBook with my display resolution set to a 1:1 of my physical screen resolution:

<img width="257" alt="Screenshot 2023-07-21 at 12 47 54 pm" src="https://github.com/Shopify/polaris/assets/612020/e394581c-ac92-41af-a910-50165b2a5d9c">

vs using the centered technique / not scaling the svg:
 
<img width="254" alt="Screenshot 2023-07-21 at 12 48 17 pm" src="https://github.com/Shopify/polaris/assets/612020/0c662029-8f5f-4139-b3e0-c995d28f87aa">

---

And here is how it looks when my display resolution set to a 1:2 of my physical screen resolution (less noticeable, but due to the slight scaling, it looks like the surrounding dark color has bled into the brighter svg color making it darker):

<img width="186" alt="Screenshot 2023-07-21 at 12 51 40 pm" src="https://github.com/Shopify/polaris/assets/612020/fbff941e-cdf1-45c1-96c2-6c91c79ebb99">


vs using the centered technique / not scaling the svg:

<img width="189" alt="Screenshot 2023-07-21 at 12 51 00 pm" src="https://github.com/Shopify/polaris/assets/612020/a39ec54f-33b5-45b9-8fde-36074b564b0a">

Reply:
This fixes a bug with the display of the checkbox. I'd avoid zooming at incredibly close levels. IMO the entire way we treat SVGs in the component is bad, we should revisit post SE23. But for now this fixes an issue we have that's pretty glaring.